dali/test/python/test_eager_operators.py

```diff
@@ -0,0 +1,196 @@
+
+
+def test_brightness_contrast_cpu():
+    compare_eager_with_pipeline('brightness_contrast')
+
+
+def test_hue_cpu():
+    compare_eager_with_pipeline('hue')
+
+
+def test_brightness_cpu():
+    compare_eager_with_pipeline('brightness')
+
+
+def test_contrast_cpu():
+    compare_eager_with_pipeline('contrast')
+
+
+def test_hsv_cpu():
+    compare_eager_with_pipeline('hsv')
+
+
+def test_color_twist_cpu():
+    compare_eager_with_pipeline('color_twist')
+
+
+def test_saturation_cpu():
+    compare_eager_with_pipeline('saturation')
+
+
+def test_shapes_cpu():
+    compare_eager_with_pipeline('shapes')
+
+
+def test_crop_cpu():
+    compare_eager_with_pipeline('crop', crop=(5, 5))
+
+
+def test_color_space_coversion_cpu():
+    compare_eager_with_pipeline('color_space_conversion',
+                                image_type=types.BGR, output_type=types.RGB)
+
+
+def test_cast_cpu():
+    compare_eager_with_pipeline('cast', dtype=types.INT32)
+
+
+def test_resize_cpu():
+    compare_eager_with_pipeline('resize', resize_x=50, resize_y=50)
+
+
+def test_gaussian_blur_cpu():
+    compare_eager_with_pipeline('gaussian_blur', window_size=5)
+
+
+def test_laplacian_cpu():
+    compare_eager_with_pipeline('laplacian', window_size=5)
+
+
+def test_crop_mirror_normalize_cpu():
+    compare_eager_with_pipeline('crop_mirror_normalize')
+
+
+def test_flip_cpu():
+    compare_eager_with_pipeline('flip', horizontal=True)
+
+
+def test_jpeg_compression_distortion_cpu():
+    compare_eager_with_pipeline('jpeg_compression_distortion', quality=10)
+
+
+def test_reshape_cpu():
+    new_shape = sample_shape.copy()
+    new_shape[0] //= 2
+    new_shape[1] *= 2
+    compare_eager_with_pipeline("reshape", shape=new_shape)
+
+
+def test_reinterpret_cpu():
+    compare_eager_with_pipeline("reinterpret", rel_shape=[0.5, 1, -1])
+
+
+def test_water_cpu():
+    compare_eager_with_pipeline("water")
+
+
+def test_sphere_cpu():
+    compare_eager_with_pipeline("sphere")
+
+
+def test_erase_cpu():
+    compare_eager_with_pipeline("erase", anchor=[0.3], axis_names="H",
+                                normalized_anchor=True, shape=[0.1], normalized_shape=True)
+
+
+def test_expand_dims_cpu():
+    compare_eager_with_pipeline("expand_dims", axes=1, new_axis_names="Z")
+
+
+def test_coord_transform_cpu():
+    M = [0, 0, 1,
+         0, 1, 0,
+         1, 0, 0]
+    compare_eager_with_pipeline("coord_transform", M=M, dtype=types.UINT8)
+
+
+def test_grid_mask_cpu():
+    compare_eager_with_pipeline("grid_mask", tile=51, ratio=0.38158387, angle=2.6810782)
+
+import torch
+
+def test_multi_paste_cpu():
+    compare_eager_with_pipeline("multi_paste", in_ids=torch.tensor([0, 1], dtype=torch.int32, device='cuda'), output_size=sample_shape)
+
+
+@raises(RuntimeError, glob=f"Argument '*' is not supported by eager operator 'crop'.")
+def _test_disqualified_argument(key):
+    tl = tensors.TensorListCPU(np.zeros((8, 256, 256, 3)))
+    eager.crop(tl, crop=[64, 64], **{key: 0})
+
```

Comment:
General question: Do we envision to add a mechanism to make sure we don't forget some operators in such tests?

Reply:
Do you mean this specific test (`test_disqualified_argument`) or adding the `test_coverage` counterpart from `test_dali_cpu_only.py` for eager operators?
For the former, at least for now it shouldn't matter - all stateless eager operators have the same set of disqualified arguments, so the selection of `crop` was arbitrary. And for the later, yes I plan on adding the coverage check when all operators will be exposed in eager API and obviously each will have its separate test.

Reply:
I meant, something to check that all operators have an eager version, unless they belong to the list of exceptions